packages/webchat/src/main.tsx

```diff
@@ -88,19 +95,30 @@ class Web extends React.Component<MainProps> {
   async load() {
     this.config = this.extractConfig()
 
+    // TODO: 6 - this implies iframe as well. Do we need this? Why?
     if (this.config.exposeStore) {
       const storePath = this.config.chatId ? `${this.config.chatId}.webchat_store` : 'webchat_store'
       set(window.parent, storePath, this.props.store)
     }
 
-    this.config.overrides && this.loadOverrides(this.config.overrides)
+    if (this.config.overrides) {
+      this.loadOverrides(this.config.overrides)
+    }
 
-    this.config.containerWidth && this.postMessageToParent('setWidth', this.config.containerWidth)
+    // TODO: 7 - again with the iframe stuff. Why do we need this.
+    if (this.config.containerWidth) {
+      this.postMessageToParent('setWidth', this.config.containerWidth)
+    }
 
-    this.config.reference && this.props.setReference!()
+    // TODO: 8 - what is reference?
+    if (this.config.reference) {
+      await this.props.setReference!()
```

Comment:
Reference is when you need to pass informations in the URL which should not be altered. The reference is a signature that certifies the payload

Reply:
Technically the reference is just used to send a "signed" message to the backend. When you set a reference on the bot's URL, we validate the signature and if it's valid, we send an event of type "session_reference" with the valid payload.

For example, imagine you have a ticketing system and you receive e-mails everytime a new ticket is created. You have 3 different tickets: tic-1, tic-2, tic-3. 

The link in each email would open the webchat on the specific ticket. If we just add a query string to the url, for example `?ticketId=1`, someone could just forge the id and use a different ID. To avoid this, you can sign the payload using the sdk, so the reference might look like `?ticketId=1=de6f53gr6g59h5yr2v1df6f9eg`. This prevents the end user to alter the message.